### PR TITLE
[CROSSDATA-552] Prefix catalogs and Zookeeper configurable paths

### PR DESCRIPTION
**Pull request by pianista215**
## Description
Allowing to set a prefix to the catalog stored in Zookeeper/MySQL/PostgreSQL. This allows you to share a catalog between different clusters, and also to have two different clusters working under the same database.

### Testing
- [x] Unit, integration tests

### Documentation
- [ ] Changelog update
- [ ] Documentation link

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/sql/crossdata/catalog/persistent/PostgreSQLXDCatalog.scala [hunk 1]
```diff
@@ -210,7 +216,7 @@ class PostgreSQLXDCatalog(sqlContext: SQLContext, override val catalystConf: Cat
     if (!resultSet.isBeforeFirst) {
       resultSet.close()
       val prepped = connection.prepareStatement(
-        s"""|INSERT INTO $db.$table (
+        s"""|INSERT INTO $db.$tableWithTableMetadata (
```
Comment (pfcoperez) [inline core/src/main/scala/org/apache/spark/sql/crossdata/catalog/persistent/PostgreSQLXDCatalog.scala]: `$db.$tableWithTableMetadata` is repeated several times, isn't it worth of be encapsulated in a method or function?

File: core/src/main/scala/org/apache/spark/sql/crossdata/daos/EphemeralTableStatusDAO.scala [hunk 2]
```diff
@@ -23,10 +23,10 @@ import org.apache.spark.sql.crossdata.models.EphemeralStatusModel
 import org.apache.spark.sql.crossdata.serializers.CrossdataSerializer
 
 trait EphemeralTableStatusDAO extends GenericDAOComponent[EphemeralStatusModel]
-with TypesafeConfigComponent with SparkLoggerComponent with CrossdataSerializer {
+with TypesafeConfigComponent with SparkLoggerComponent with CrossdataSerializer with PrefixedDAO {
 
   override implicit val formats = json4sJacksonFormats
 
-  override val dao: DAO = new GenericDAO(Option(EphemeralTableStatusPath))
+  override val dao: DAO = new GenericDAO(Option(prefix+EphemeralTableStatusPath))
```
Comment (pfcoperez) [inline core/src/main/scala/org/apache/spark/sql/crossdata/daos/EphemeralTableStatusDAO.scala]: Same as above, it is not a matter of saving space or the like. Just that if your future yourself - o some other guy (or girl) - need to change the way the name is built then you, he or she will suffer while hammering the change all over the code.

**Comment by pfcoperez** [comment]
+1

**Comment by darroyocazorla** [comment]
+1

**Pull request merged**